<!SPECIAL_ANNOTATION_NAME_SHOULD_BE_DENOTED_BY!>@file:MyJvmName("Test")<!>
<!SPECIAL_ANNOTATION_NAME_SHOULD_BE_DENOTED_BY!>@file:MyJvmMultifileClass<!>
package test

import kotlin.jvm.JvmName as MyJvmName
import kotlin.jvm.JvmMultifileClass as MyJvmMultifileClass